### DISCUSSION TITLE
[firebase_crashlytic] Return Future<bool> as result of method calls to send reports to crashlytic

### DISCUSSION THREAD

**Discussion by suesitran** [firebase/flutterfire#9461] [Feature request]
Currently all Crashlytic methods to send report to Crashlytic return `Future<void>`, and developers do not know when it fails to send report. So I'm suggesting to return a `Future<bool>` value, `true` means success, and `false` means error, and if possible, give details of the error as well.

Methods affected: 
- recordError
- recordFlutterError
- recordFlutterFatalError
- sendUnsentReports

**Comment by radvansky-tomas** [comment]
that does not seem consistent. It should still be future<void> but it should throw respective firebase error type.

Did u actually try to capture error there ?

**Reply by suesitran** [reply]
There's error captured for full app using `runZonedGuarded`, but I do not see any error here. Do I have to do try/catch on each call?

**Reply by suesitran** [reply]
and yes, it should still be future, instead of `Future<void>`, can it be `Future<bool>`?
